torch/nn/modules/module.py

```diff
@@ -1642,10 +1642,24 @@ def _call_impl(self, *args, **kwargs):
 
     def __getstate__(self):
         state = self.__dict__.copy()
+
+        # Support for slots in child classes
+        if getattr(self, "__slots__", None):
```

Comment:
I had missed something in my previous pass of the PR, we don't only want to save the `__slots__` defined in the current class, but if this class subclasses other classes that subclass `torch.nn.Module`, we want to get their `__slots__`, which we can do using `copyreg._slotnames`. See below for an example illustrating this:

e.g. 

```python
class Block(torch.nn.Module):
    __slots__ = ['b']
    def __init__(self):
        super().__init__()
        self.b = 3

class DummyModel(Block):
    __slots__ = ['a']
    def __init__(self):
        super().__init__()
        self.a = 2


m = DummyModel()
print(m.__slots__)  # ['a']
print(copyreg._slotnames(m.__class__))  # ['a', 'b']

```

This shows how we do something similar for `torch.Tensor`

https://github.com/pytorch/pytorch/blob/fb288aa99b1c7c1792a3f90396d2ab7275eab2d9/torch/_tensor.py#L202-L205